code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java

```diff
@@ -143,4 +149,29 @@ public Instance getInstanceByUuidOrExternalId(Long accountId, String uuid, Strin
                 .fetchInto(InstanceRecord.class);
     }
 
+    @Override
+    public Long getInstanceHostId(long instanceId) {
+        List<? extends InstanceHostMap> maps = mapDao.findNonRemoved(InstanceHostMap.class, Instance.class,
+                instanceId);
+        if (!maps.isEmpty()) {
```

Comment:
nit: its easier to read code when the first block of a if statement is the true case.  so in this situation.

```java
if (maps.isEmpty()) {
  return null
}

return maps.get(0).getHostId();
```